lib/buffer.js

```diff
@@ -272,19 +272,66 @@ Buffer.concat = function(list, length) {
 };
 
 
+function base64ByteLength(str) {
+  var len = str.length;
+  var bytes = len;
+
+  // Handle padding
+  if (str[len - 1] === '=')
+    bytes--;
+  if (len > 2 && str[len - 2] === '=')
+    bytes--;
+
+  // Base64 ratio: 3/4
+  bytes = (bytes / 4) * 3;
```

Comment:
I can think of two reasons why `bytes >>> 2` might be preferable to Math.floor(), apart from performance reasons:

1. There is no risk of future changes accidentally letting floating point values slip through, and

2. It's robust in the face of people monkey-patching Math.floor.